File: common/src/main/java/com/ssblur/scriptor/registry/words/Actions.java
package com.ssblur.scriptor.registry.words;

import com.ssblur.scriptor.api.word.Action;
import com.ssblur.scriptor.item.ScriptorItems;
import com.ssblur.scriptor.word.action.*;
import com.ssblur.scriptor.word.action.bound.BoundSwordAction;
import com.ssblur.scriptor.word.action.bound.BoundToolAction;
import com.ssblur.scriptor.word.action.teleport.BringAction;
import com.ssblur.scriptor.word.action.teleport.GotoAction;
import com.ssblur.scriptor.word.action.teleport.SwapAction;
import net.minecraft.core.registries.Registries;
import net.minecraft.resources.ResourceLocation;
import net.minecraft.tags.TagKey;

import java.util.List;

@SuppressWarnings("unused")
public class Actions {
  private static final WordRegistry INSTANCE = WordRegistry.INSTANCE;

  public final Action INFLAME = INSTANCE.register("inflame", new InflameAction());
  public final Action LIGHT = INSTANCE.register("light", new LightAction());
  public final Action HEAL = INSTANCE.register("heal", new HealAction());
  public final Action SMITE = INSTANCE.register("smite", new SmiteAction());
  public final Action EXPLOSION = INSTANCE.register("explosion", new ExplosionAction());
  public final Action GOTO = INSTANCE.register("goto", new GotoAction());
  public final Action SWAP = INSTANCE.register("swap", new SwapAction());
  public final Action BRING = INSTANCE.register("bring", new BringAction());
  public final Action BREAK = INSTANCE.register("break", new BreakBlockAction());
  public final Action PLACE = INSTANCE.register("place", new PlaceBlockAction());
  public final Action HARM = INSTANCE.register("harm", new HarmAction());
  public final Action COLOR = INSTANCE.register("color", new ColorAction());
  public final Action TIME = INSTANCE.register("time", new AdvanceTimeAction());
  public final Action CLEAR_WEATHER = INSTANCE.register("clear_weather", new ClearWeatherAction());
  public final Action RAIN = INSTANCE.register("rain", new RainAction());

  public final Action BOUND_SWORD = INSTANCE.register("bound_sword", new BoundSwordAction());
  public final Action BOUND_AXE = INSTANCE.register("bound_axe",
    new BoundToolAction(ScriptorItems.BOUND_AXE, List.of(TagKey.create(Registries.BLOCK, ResourceLocation.withDefaultNamespace("mineable/axe")))));
  public final Action BOUND_SHOVEL = INSTANCE.register("bound_shovel",
    new BoundToolAction(ScriptorItems.BOUND_SHOVEL, List.of(TagKey.create(Registries.BLOCK, ResourceLocation.withDefaultNamespace("mineable/shovel")))));
  public final Action BOUND_PICKAXE = INSTANCE.register("bound_pickaxe",
    new BoundToolAction(ScriptorItems.BOUND_PICKAXE, List.of(TagKey.create(Registries.BLOCK, ResourceLocation.withDefaultNamespace("mineable/pickaxe")))));
}

File: common/src/main/kotlin/com/ssblur/scriptor/registry/words/Actions.kt
package com.ssblur.scriptor.registry.words

import com.ssblur.scriptor.item.ScriptorItems
import com.ssblur.scriptor.registry.words.WordRegistry.register
import com.ssblur.scriptor.word.action.*
import com.ssblur.scriptor.word.action.bound.BoundSwordAction
import com.ssblur.scriptor.word.action.bound.BoundToolAction
import com.ssblur.scriptor.word.action.teleport.BringAction
import com.ssblur.scriptor.word.action.teleport.GotoAction
import com.ssblur.scriptor.word.action.teleport.SwapAction
import net.minecraft.core.registries.Registries
import net.minecraft.resources.ResourceLocation
import net.minecraft.tags.TagKey
import java.util.List

@Suppress("unused")
object Actions {
    val INFLAME = register("inflame", InflameAction())
    val LIGHT = register("light", LightAction())
    val HEAL = register("heal", HealAction())
    val SMITE = register("smite", SmiteAction())
    val EXPLOSION = register("explosion", ExplosionAction())
    val GOTO = register("goto", GotoAction())
    val SWAP = register("swap", SwapAction())
    val BRING = register("bring", BringAction())
    val BREAK = register("break", BreakBlockAction())
    val PLACE = register("place", PlaceBlockAction())
    val HARM = register("harm", HarmAction())
    val COLOR = register("color", ColorAction())
    val TIME = register("time", AdvanceTimeAction())
    val CLEAR_WEATHER = register("clear_weather", ClearWeatherAction())
    val RAIN = register("rain", RainAction())

    val BOUND_SWORD = register("bound_sword", BoundSwordAction())
    val BOUND_AXE = register(
        "bound_axe",
        BoundToolAction(
            ScriptorItems.BOUND_AXE,
            List.of(TagKey.create(Registries.BLOCK, ResourceLocation.withDefaultNamespace("mineable/axe")))
        )
    )
    val BOUND_SHOVEL = register(
        "bound_shovel",
        BoundToolAction(
            ScriptorItems.BOUND_SHOVEL,
            List.of(TagKey.create(Registries.BLOCK, ResourceLocation.withDefaultNamespace("mineable/shovel")))
        )
    )
    val BOUND_PICKAXE = register(
        "bound_pickaxe",
        BoundToolAction(
            ScriptorItems.BOUND_PICKAXE,
            List.of(TagKey.create(Registries.BLOCK, ResourceLocation.withDefaultNamespace("mineable/pickaxe")))
        )
    )
}
